services/blockchain-app-registry/shared/utils/fsUtils.js

```diff
@@ -179,6 +179,40 @@ const rename = async (oldName, newName) => new Promise((resolve, reject) => {
 	});
 });
 
+const deleteFolderIfEmpty = async (folderPath) => {
+	const files = await fs.promises.readdir(folderPath);
+
+	if (files.length === 0) {
+		await fs.promises.rmdir(folderPath);
+		logger.trace(`Deleted folder: ${folderPath}`);
+	}
+};
+
+const deleteEmptyFoldersAndNonMetaFiles = async (folderPath) => {
+	const files = await fs.promises.readdir(folderPath);
+
+	for (let i = 0; i < files.length; i++) {
+		/* eslint-disable no-await-in-loop */
+		const file = files[i];
+		const filePath = path.join(folderPath, file);
+		const isDirectory = (await fs.promises.lstat(filePath)).isDirectory();
+
+		if (isDirectory) {
+			await deleteEmptyFoldersAndNonMetaFiles(filePath);
+			await deleteFolderIfEmpty(filePath);
+		} else {
+			const fileName = path.basename(filePath);
+
+			if (!ALLOWED_FILE_EXTENSIONS.some((ending) => file.endsWith(ending))
+					&& !ALLOWED_FILES.some((name) => fileName === name)) {
```

Comment:
Can we reuse this method to avoid duplicate logic. Currently we use this method while syncing with new commit in remote repo to determine if we should download the added/updated file. So, currently we are not downloading the added/updated logo image. Could we please update the logic so that we consider both meta files and extensions?
https://github.com/LiskHQ/lisk-service/blob/feature/1157-prepare-lisk-service-v070-for-beta/services/blockchain-app-registry/shared/utils/downloadRepository.js#L201-L203

Reply:
Confirmed with @sameersubudhi, not to modify code from download repository. Maybe better to create a separate issue for this.

Reply:
The existing piece of code can be left as is and requires no change.